api/controllers/HomeController.js

```diff
@@ -31,18 +31,23 @@ module.exports = {
     });
   },
 
-  reference: function(req, res) {
-    User.findOne({name: req.params.user}).exec(function (err, user){
-      if (user) {
-        res.view();
-      } else {
-        res.view('404', {data: {user: req.params.user, error: "User not found"}});
+  reference: async function(req, res) {
+    try {
+      return res.view({refUser: await Users.get(req.user, req.params.user)});
```

Comment:
We use the `user` ejs local to describe to describe the currently authenticated user, as we did before. However, `user` no longer contains the user's references. Instead, we use `refUser` for the user whose references are being viewed. (On the index page, `refUser.name === user.name`.